cmd/rendmp/src/blackbox.rs

```diff
@@ -0,0 +1,774 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt::{Display, Formatter};
+use zerocopy::FromBytes;
+
+type U32 = zerocopy::byteorder::U32<zerocopy::BE>;
+type U16 = zerocopy::byteorder::U16<zerocopy::BE>;
+type I16 = zerocopy::byteorder::I16<zerocopy::BE>;
+
+fn format_faults<T: Into<u32> + std::fmt::Binary + Copy>(
+    v: T,
+    faults: &[(u32, &'static str)],
+    f: &mut Formatter,
+) -> std::fmt::Result {
+    let u: u32 = v.into();
+    let faults: Vec<String> = faults
+        .iter()
+        .filter(|(i, _)| u & (1 << i) != 0)
+        .map(|(_, v)| v.to_string())
```

Comment:
Two tiny nits - maybe `filter_map()` instead of `.filter(..).map(..)`, and I don't think `.to_string()` is necessary since we already have a `&str`. Combining both:

```rust
     let faults: Vec<_> = faults
         .iter()
         .filter_map(|&(i, v)| (u & (1 << i) != 0).then_some(v))
         .collect();
```

Reply:
Neat!